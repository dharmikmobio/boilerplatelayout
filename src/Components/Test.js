/* eslint-disable no-unused-vars */
import React,{useState,startTransition} from 'react'
import Boiler from './Boiler'


const Test = () => {
    const [name, setName] = useState("React 18");
    // const [email, setEmail] = useState("testing@gmail.com");
  
    const updateAll = () =>{
        setName("React new version");
        
        // startTransition(()=>{
        //     setEmail("test@gmail.com");
        // })
        
    }
    console.warn("update");

    const [gender, setGender] = useState("male");
    const secoundAll = (e) =>{
        setGender(e.target.getAttribute("data-gender"));
    }
  return (
    <>
        <Boiler>
            <h1 className='new'>React 18</h1>
            <p className='new2'>{name}</p>
            {/* <p className='new2'>{email}</p> */}
            <button className='btn' onClick={updateAll}>Update All </button>


            <button data-gender="male" onClick={secoundAll}>Male </button>
            <button data-gender="female" onClick={secoundAll}>Female </button>
            <br/>
            <div className={`box ${gender}`} style={{margin:"10px"}}>
                select{gender}
            </div>
            
        </Boiler>
    </>
  )
}

export default Test