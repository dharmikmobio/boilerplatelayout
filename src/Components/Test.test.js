import React from 'react';
import Test from './Test';

import { shallow } from 'enzyme';



describe("shallow test page",()=>{
    it('Object check',()=>{
        let wrapper = shallow(<Test/>);
        //check all component inside
        // console.log(wrapper);
        // console.log(wrapper.debug());
        //check class in component
        expect(wrapper.exists('.btn')).toEqual(true);
    })


// it('state check only class components',()=>{
//     let wrapper = shallow(<Test/>);
//     expect(wrapper.state()).toEqual({name:""});
// })

    it('html check',()=>{
        let wrapper = shallow(<Test/>);
        expect(wrapper.contains(<h1 className='new'>React 18</h1>)).toEqual(true);
    })

    it('Only msj check',()=>{
        let wrapper = shallow(<Test/>);
        expect(wrapper.containsAllMatchingElements([<h1>React 18</h1>])).toEqual(true);
    })

    it('find check',()=>{
        let wrapper = shallow(<Test/>);
        expect(wrapper.find('.new').length).toBe(1);
    })

    it('snapshot 1 check',()=>{
        let wrapper = shallow(<Test/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('function check in function component',()=>{
        const wrapper = shallow(<Test/>);
        wrapper.find(".btn").simulate("click");
        // console.log(wrapper.find('.new2').text());
        expect(wrapper.find('.new2').text()).toBe("React new version")
    })

    it('gender state by click function',()=>{
        const wrapper = shallow(<Test/>);
        const btn = wrapper.find("button").last();

        btn.simulate('click',{
            target:{
                getAttribute:function(){
                    return btn.props()['data-gender']
                }
            }
        })
        expect(wrapper.find('.box.female').length).toEqual(1);
        
    })



})







