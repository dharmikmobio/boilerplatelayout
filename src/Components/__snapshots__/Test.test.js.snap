// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow test page snapshot 1 check 1`] = `
<Fragment>
  <Boiler>
    <h1
      className="new"
    >
      React 18
    </h1>
    <p
      className="new2"
    >
      React 18
    </p>
    <button
      className="btn"
      onClick={[Function]}
    >
      Update All 
    </button>
    <button
      data-gender="male"
      onClick={[Function]}
    >
      Male 
    </button>
    <button
      data-gender="female"
      onClick={[Function]}
    >
      Female 
    </button>
    <br />
    <div
      className="box male"
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      select
      male
    </div>
  </Boiler>
</Fragment>
`;
